clients/js/src/index.ts

```diff
@@ -39,3 +39,61 @@ export class CacReader {
         return deepMerge(targetConfig, ...requiredOverrides);
     }
 }
+
+export class ExperimentReader {
+    experiments: ExperimentStore;
+
+    constructor(experiments: Experiments) {
+        this.experiments = new Map(experiments.map(exp => [exp.id, exp]));
+    }
+
+    public getApplicableVariant(data: IObject, toss: number): Array<String> {
+        const experiments = this.getSatisfiedExperiments(data);
+        const variants = [];
+        for (const exp of experiments) {
+            const v = this.decideVariant(exp.traffic_percentage, exp.variants, toss);
+            if (v) {
+                variants.push(v.id)
+            }
+        }
+        return variants;
+    };
+
+    getSatisfiedExperiments(data: IObject): Experiments {
+        const experiments = [];
+        const iter = this.experiments.values();
+
+        while (true) {
+            let result = iter.next();
+            if (result.done) break;
+            if (jsonLogic.apply(result.value.context, data)) {
+                experiments.push(result.value);
+            }
+        }
+
+        return experiments;
+    };
+
+    // decide which variant to return among all applicable experiments
+    decideVariant(
+        traffic: number,
```

Comment:
the type `number` encompasses decimal, float, hex, int, etc in JS. Can we check that all numbers provided are a positive integer?

Reply:
we cannot add restriction of positive integers only, to test EXPERIMENT variant we need to send -1

Reply:
Sure, you can the allowed range as `(-1, 100)`